nmt.go

```diff
@@ -277,6 +367,14 @@ func (n *NamespacedMerkleTree) calculateAbsenceIndex(nID namespace.ID) int {
 	panic("calculateAbsenceIndex() called although (nID < minNID) or (maxNID < nID) for provided nID")
 }
 
+// foundInRange returns a tuple of (true, startIndex, endIndex) if a range of leaves
+// in the namespace tree is found with namespace IDs that match the given nID.
+// The startIndex and endIndex indicate the starting and ending indices of the range, respectively.
+//
+// If no leaves are found, foundInRange returns (false, 0, 0).
+//
+// The endIndex is non-inclusive,
+// meaning it does not include the leaf at that index in the range.
 func (n *NamespacedMerkleTree) foundInRange(nID namespace.ID) (bool, int, int) {
```

Comment:
[no change needed][optional] we can remove some of the godoc comment by using named return values https://go.dev/tour/basics/7 but that's a debated topic on our team so it may not make sense to adopt for this isolated use-case.

```suggestion
func (n *NamespacedMerkleTree) foundInRange(nID namespace.ID) (found bool, startIndex int, endIndex int) {
```

Reply:
Thanks, incorporated your suggestion https://github.com/celestiaorg/nmt/commit/26b1ea676ac264bb2c6894efcd7a4f4aaf3924b3